components/Tabs/tests/unit/Theming/TabsThemingTests.m

```diff
@@ -0,0 +1,185 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import <MaterialComponents/MaterialColorScheme.h>
+#import <MaterialComponents/MaterialTabs.h>
+#import <MaterialComponents/MaterialTypographyScheme.h>
+#import <MaterialComponentsBeta/MaterialContainerScheme.h>
+#import <MaterialComponentsBeta/MaterialTabs+Theming.h>
+
+static const CGFloat kUnselectedTitleOpacity = (CGFloat)0.6;
+static const CGFloat kUnselectedImageOpacity = (CGFloat)0.54;
+static const CGFloat kBottomDividerOpacity = (CGFloat)0.12;
+
+@interface MDCTabsThemingTest : XCTestCase
+@property(nonatomic, strong) MDCTabBar *tabBar;
+@property(nonatomic, strong) MDCSemanticColorScheme *colorScheme;
+@property(nonatomic, strong) MDCTypographyScheme *typographyScheme;
+@property(nonatomic, strong) MDCContainerScheme *containerScheme;
+@end
+
+@implementation MDCTabsThemingTest
+
+- (void)setUp {
+  [super setUp];
+
+  self.tabBar = [[MDCTabBar alloc] init];
+  self.colorScheme =
+      [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
+  self.typographyScheme =
+      [[MDCTypographyScheme alloc] initWithDefaults:MDCTypographySchemeDefaultsMaterial201804];
+  self.containerScheme = [[MDCContainerScheme alloc] init];
```

Comment:
`colorScheme` and `typographyScheme` were not assigned here intentionally, since we want to test the case when no specific scheme is included in the `containerScheme`.

Reply:
The MDCContainerScheme has defaults for `colorScheme` and `typographyScheme`.  They are the same defaults you're using here.

Reply:
Right, that's exactly what the Defaults test wants to test, otherwise `if (!containerScheme.colorScheme)` line won't get test coverage.

Reply:
There is no code path that could execute `if(!containerScheme.colorScheme)` that property is always `nonnull`.

Reply:
Ah I forgot the `nonnull` changes and was trapped when looking at some other theming extension implementation. Addressed in 07a3eca6e7891ca4522fd955583df447eeb6fb54.